src/org/zaproxy/zap/extension/technology/TechnologyAPI.java

```diff
@@ -0,0 +1,158 @@
+	
+	private static final String ACTION_INCLUDE_TECHS = "includeContextTechnologies";
+	private static final String ACTION_INCLUDE_ALL_TECHS = "includeAllContextTechnologies";
+	private static final String ACTION_EXCLUDE_TECHS = "excludeContextTechnologies";
+	private static final String ACTION_EXCLUDE_ALL_TECHS = "excludeAllContextTechnologies";
+	
+	private static final String PARAM_CONTEXT_ID = "contextId";
+	private static final String PARAM_TECH_NAMES = "technologyNames";
+	
+	public TechnologyAPI() {
+		String[] contextId = new String[] { PARAM_CONTEXT_ID };
+		String[] contextIdAndTechNames = new String[] { PARAM_CONTEXT_ID, PARAM_TECH_NAMES };
+		
+		this.addApiView(new ApiView(VIEW_ALL_TECHS));
+		this.addApiView(new ApiView(VIEW_INCLUDED_TECHS, contextId));
+		this.addApiView(new ApiView(VIEW_EXCLUDED_TECHS, contextId));
+		
+		this.addApiAction(new ApiAction(ACTION_INCLUDE_TECHS, contextIdAndTechNames));
+		this.addApiAction(new ApiAction(ACTION_INCLUDE_ALL_TECHS, contextId));
+		this.addApiAction(new ApiAction(ACTION_EXCLUDE_TECHS, contextIdAndTechNames));
+		this.addApiAction(new ApiAction(ACTION_EXCLUDE_ALL_TECHS, contextId));
+	}
+	
+	@Override
+	public String getPrefix() {
+		return PREFIX;
+	}
+	
+	@Override
+	public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+		log.debug("handleApiView " + name + " " + params.toString());
+		
+		ApiResponseList resultList;
+		TechSet techSet;
+		
+		switch(name) {
+		case VIEW_ALL_TECHS:
+			resultList = new ApiResponseList(name);
+			for(Tech tech : Tech.builtInTech) {
+				resultList.addItem(new ApiResponseElement(TECH_NAME, tech.toString()));
+			}
+			return resultList;
+		case VIEW_INCLUDED_TECHS:
+			resultList = new ApiResponseList(name);
+			techSet = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID).getTechSet();
+			for (Tech tech : techSet.getIncludeTech()) {
+				resultList.addItem(new ApiResponseElement(TECH_NAME, tech.toString()));
+			}
+			return resultList;
+		case VIEW_EXCLUDED_TECHS:
+			resultList = new ApiResponseList(name);
+			techSet = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID).getTechSet();
+			for(Tech tech : techSet.getExcludeTech()) {
+				resultList.addItem(new ApiResponseElement(TECH_NAME, tech.toString()));
+			}
+			return resultList;
+		default:
+			throw new ApiException(Type.BAD_VIEW);
+		}
+	}
+	
+	@Override
+	public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+		log.debug("handleApiAction " + name + " " + params.toString());
+		
+		TechSet techSet = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID).getTechSet();
+		String[] techNames;
+		switch(name) {
+		case ACTION_INCLUDE_TECHS:
+			techNames = getParam(params, PARAM_TECH_NAMES, "").split(",");
+			for(String techName : techNames) {
+				techSet.include(getTech(techName));
+			}
+			break;
+		case ACTION_INCLUDE_ALL_TECHS:
+			techSet = TechSet.AllTech;
```

Comment:
This statement has not effect to the context. It should be used the method Context.setTechSet(TechSet).